v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerLTBase.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import static java.util.Arrays.stream;
+import static org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.cloud.datastream.v1.DestinationConfig;
+import com.google.cloud.datastream.v1.SourceConfig;
+import com.google.cloud.datastream.v1.Stream;
+import com.google.common.io.Resources;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.apache.beam.it.gcp.TemplateLoadTestBase;
+import org.apache.beam.it.gcp.datastream.DatastreamResourceManager;
+import org.apache.beam.it.gcp.datastream.JDBCSource;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.beam.it.gcp.spanner.SpannerResourceManager;
+import org.apache.beam.it.gcp.storage.GcsResourceManager;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Strings;
+
+/**
+ * Base class for DataStreamToSpanner Load tests. It provides helper functions related to
+ * environment setup and assertConditions.
+ */
+public class DataStreamToSpannerLTBase extends TemplateLoadTestBase {
+
+  /**
+   * Helper function for creating all datastream resources required by DataStreamToSpanner template.
+   * Source connection profile, Destination connection profile, Stream. And then Starts the stream.
+   *
+   * @param artifactBucketName
+   * @param gcsPrefix
+   * @param jdbcSource
+   * @param datastreamResourceManager
+   * @return created stream
+   */
+  public Stream createDatastreamResources(
+      String artifactBucketName,
+      String gcsPrefix,
+      JDBCSource jdbcSource,
+      DatastreamResourceManager datastreamResourceManager) {
+    SourceConfig sourceConfig =
+        datastreamResourceManager.buildJDBCSourceConfig("mysql-profile", jdbcSource);
+
+    // Create Datastream GCS Destination Connection profile and config
+    DestinationConfig destinationConfig =
+        datastreamResourceManager.buildGCSDestinationConfig(
+            "gcs-profile",
+            artifactBucketName,
+            gcsPrefix,
+            DatastreamResourceManager.DestinationOutputFormat.AVRO_FILE_FORMAT);
+
+    // Create and start Datastream stream
+    Stream stream =
+        datastreamResourceManager.createStream("stream1", sourceConfig, destinationConfig);
```

Comment:
These names should be randomized and auto-generated.

Reply:
These are just prefixes. The resourcemanagers take care of creating a unique random ids.